packages/server/src/http/resolveHTTPResponse.ts

```diff
@@ -138,66 +257,88 @@ export async function resolveHTTPResponse<
+      const transformedJSON = transformTRPCResponse(
+        router._def._config,
+        result,
+      );
+      const body = JSON.stringify(transformedJSON);
+      const chunk: ResponseChunk = [-1, body];
+      onChunk?.(chunk);
+
+      return {
+        status: headResponse.status,
+        headers: headResponse.headers,
+        body,
+      };
+    } else {
+      /**
+       * Streaming response:
+       * - block on none, call `onChunk` as soon as each response is ready
+       * - create headers with minimal data (cannot know the response body in advance)
+       * - return void
+       */
+
+      try {
+        const headResponse = initResponse({ctx, paths, type, responseMeta: opts.responseMeta});
+        onHead(headResponse);
+
+        const indexedPromises = new Map(promises.map((promise, index) => [index, promise.then(r => [index, r] as const)]));
+        for (let i = 0; i < paths.length; i++) {
+          const [index, untransformedJSON] = await Promise.race(indexedPromises.values());
+          indexedPromises.delete(index);
 
-    const rawResults = await Promise.all(
-      paths.map(async (path, index) => {
-        const input = inputs[index];
-
-        try {
-          const output = await callProcedure({
-            procedures: router._def.procedures,
-            path,
-            rawInput: input,
-            ctx,
-            type,
-          });
-          return {
-            input,
-            path,
-            data: output,
-          };
-        } catch (cause) {
-          const error = getTRPCErrorFromUnknown(cause);
-
-          opts.onError?.({ error, path, input, ctx, type: type, req });
-          return {
-            input,
-            path,
-            error,
-          };
+          const transformedJSON = transformTRPCResponse(
+            router._def._config,
+            untransformedJSON,
+          );
+          const body = JSON.stringify(transformedJSON);
+
+          const chunk: ResponseChunk = [index, body];
+          onChunk(chunk);
         }
-      }),
-    );
-    const errors = rawResults.flatMap((obj) => (obj.error ? [obj.error] : []));
-    const resultEnvelopes = rawResults.map((obj): TRouterResponse => {
-      const { path, input } = obj;
-
-      if (obj.error) {
-        return {
-          error: getErrorShape({
-            config: router._def._config,
-            error: obj.error,
-            type,
-            path,
-            input,
-            ctx,
-          }),
-        };
-      } else {
-        return {
-          result: {
-            data: obj.data,
-          },
-        };
+      } catch (cause) {
+        // at this point it's too late to fail gracefully
+        // because headers were sent and stream has started
+        // TODO: we might imagine some protocol using a custom HTTP Trailer to send errors after streaming
+        console.error('Unexpected error in streaming response', cause);
```

Comment:
Not sure what would be the best way to handle this error. At this point, streaming has started, headers went off, there's nothing more to be done.

- The error can only come from `transformTRPCResponse` (so basically, the `.serialize()` function).
- client-side, missing procedure results in a stream are handled (rejected)

Possible improvements are
- log the error client-side by passing it as a `Trailer` after the stream (not sure how useful that is, and it would add client-side code)
- add a `try/catch` inside the `for` loop so that 1 failure of `.serialize` doesn't prevent the entire stream (on a regular batch response, 1 failure fails the entire batch)